test/test_qat.py

```diff
@@ -123,17 +122,40 @@ def ref_op(x):
             else:
                 ref_op = compose([conv_op, bn_op, relu_op])
 
-            result_ref = ref_op(input)
-            result_actual = qat_op(input)
-            self.assertEqual(result_ref, result_actual)
+            input_clone = input.clone().detach().requires_grad_()
+            for i in range(3):
+                result_ref = ref_op(input)
+                result_actual = qat_op(input_clone)
+                self.assertEqual(result_ref, result_actual)
+
+                # backward
+                dout = torch.randn(result_ref.size(), dtype=torch.float)
+                loss = (result_ref - dout).sum()
+                loss.backward()
+                input_grad_ref = input.grad.cpu()
+                weight_grad_ref = conv_op.weight.grad.cpu()
+                gamma_grad_ref = bn_op.weight.grad.cpu()
+                beta_grad_ref = bn_op.bias.grad.cpu()
+                running_mean_ref = bn_op.running_mean
+                running_var_ref = bn_op.running_var
+                num_batches_tracked_ref = bn_op.num_batches_tracked
+                loss = (result_actual - dout).sum()
+                loss.backward()
+                input_grad_actual = input_clone.grad.cpu()
+                weight_grad_actual = qat_op.weight.grad.cpu()
+                gamma_grad_actual = qat_op.gamma.grad.cpu()
+                beta_grad_actual = qat_op.beta.grad.cpu()
+                running_mean_actual = qat_op.running_mean
+                running_var_actual = qat_op.running_var
+                num_batches_tracked_actual = qat_op.num_batches_tracked
+                self.assertEqual(input_grad_ref, input_grad_actual)
+                self.assertEqual(weight_grad_ref, weight_grad_actual, prec=1e-4)
+                self.assertEqual(gamma_grad_ref, gamma_grad_actual, prec=1e-4)
+                self.assertEqual(beta_grad_ref, beta_grad_actual)
+                self.assertEqual(num_batches_tracked_ref, num_batches_tracked_actual)
+                self.assertEqual(running_mean_ref, running_mean_actual)
+                # self.assertEqual(running_var_ref, running_var_actual, prec=1e-2)
```

Comment:
The problem is due to incorrect calculation of variance. Batch norm code at: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L223 uses unbiased variance estimate, while the convbn code does not:

https://github.com/pytorch/pytorch/blob/3c5f7c2e4656c84a79168be4e0b7edd8d41e5e2e/torch/nn/_intrinsic/qat/modules/conv_fused.py#L122
Please change:
```
          batch_var = torch.var(conv_orig, dim=[0, 2, 3], unbiased=False)
```
To:
```
          batch_var = torch.var(conv_orig, dim=[0, 2, 3], unbiased=True)
```
  and check if the mismatch in running_var is gone.